src/irr/asset/CGLILoader.cpp

```diff
@@ -0,0 +1,552 @@
+				///case gl::INTERNAL_RGB32U: return getTranslatedFinalFormat(EF_R32G32B32_UINT);			//GL_RGB32UI
+				case gl::INTERNAL_RGBA32U: return getTranslatedFinalFormat(EF_R32G32B32A32_UINT);			//GL_RGBA32UI
+
+				case gl::INTERNAL_RGB10A2U: return getTranslatedFinalFormat(EF_A2R10G10B10_UINT_PACK32);			//GL_RGB10_A2UI
+				case gl::INTERNAL_RGB10A2I_EXT: return getTranslatedFinalFormat(EF_A2R10G10B10_SINT_PACK32);	//GL_RGB10_A2I
+
+				// signed integer formats
+				case gl::INTERNAL_R8I: return getTranslatedFinalFormat(EF_R8_SINT);				//GL_R8I
+				case gl::INTERNAL_RG8I: return getTranslatedFinalFormat(EF_R8G8_SINT);				//GL_RG8I
+				case gl::INTERNAL_RGB8I: return getTranslatedFinalFormat(EF_R8G8B8_SINT);			//GL_RGB8I
+				case gl::INTERNAL_RGBA8I: return getTranslatedFinalFormat(EF_R8G8B8A8_SINT);			//GL_RGBA8I
+
+				case gl::INTERNAL_R16I: return getTranslatedFinalFormat(EF_R16_SINT);				//GL_R16I
+				case gl::INTERNAL_RG16I: return getTranslatedFinalFormat(EF_R16G16_SINT);			//GL_RG16I
+				///case gl::INTERNAL_RGB16I: return getTranslatedFinalFormat(EF_R16G16B16_SINT);			//GL_RGB16I
+				case gl::INTERNAL_RGBA16I: return getTranslatedFinalFormat(EF_R16G16B16A16_SINT);			//GL_RGBA16I
+
+				case gl::INTERNAL_R32I: return getTranslatedFinalFormat(EF_R32_SINT);				//GL_R32I
+				case gl::INTERNAL_RG32I: return getTranslatedFinalFormat(EF_R32G32_SINT);			//GL_RG32I
+				///case gl::INTERNAL_RGB32I: return getTranslatedFinalFormat(EF_R32G32B32_SINT);			//GL_RGB32I
+				case gl::INTERNAL_RGBA32I: return getTranslatedFinalFormat(EF_R32G32B32A32_SINT);			//GL_RGBA32I
+
+				// Floating formats
+				case gl::INTERNAL_R16F: return getTranslatedFinalFormat(EF_R16_SFLOAT);				//GL_R16F
+				case gl::INTERNAL_RG16F: return getTranslatedFinalFormat(EF_R16G16_SFLOAT);			//GL_RG16F
+				///case gl::INTERNAL_RGB16F: return getTranslatedFinalFormat(EF_R16G16B16_SFLOAT);			//GL_RGB16F
+				case gl::INTERNAL_RGBA16F: return getTranslatedFinalFormat(EF_R16G16B16A16_SFLOAT);			//GL_RGBA16F
+
+				case gl::INTERNAL_R32F: return getTranslatedFinalFormat(EF_R32_SFLOAT);				//GL_R32F
+				case gl::INTERNAL_RG32F: return getTranslatedFinalFormat(EF_R32G32_SFLOAT);			//GL_RG32F
+				///case gl::INTERNAL_RGB32F: return getTranslatedFinalFormat(EF_R32G32B32_SFLOAT);			//GL_RGB32F
+				case gl::INTERNAL_RGBA32F: return getTranslatedFinalFormat(EF_R32G32B32A32_SFLOAT);			//GL_RGBA32F
+
+				case gl::INTERNAL_R64F_EXT: return getTranslatedFinalFormat(EF_R64_SFLOAT);			//GL_R64F
+				case gl::INTERNAL_RG64F_EXT: return getTranslatedFinalFormat(EF_R64G64_SFLOAT);		//GL_RG64F
+				///case gl::INTERNAL_RGB64F_EXT: return getTranslatedFinalFormat(EF_R64G64B64_SFLOAT);		//GL_RGB64F
+				case gl::INTERNAL_RGBA64F_EXT: return getTranslatedFinalFormat(EF_R64G64B64A64_SFLOAT);		//GL_RGBA64F
+
+				// sRGB formats
+				case gl::INTERNAL_SR8: return getTranslatedFinalFormat(EF_R8_SRGB);				//GL_SR8_EXT
```

Comment:
This also concerns #412 , if you could add a utility that expands a R8_SRBG texture into a RGB8_SRGB texture to ImageLoaderBase (or whatever that common class with hidden utilities is called) and use this in GLI and TGA loader whenever an R8_SRGB format is encountered.

The reasoning for this is that the R8 format with SRGB encoding used to be available in compatibility (with 2.1) OpenGL profile, but got removed from 3.3 core profile only to be added back to OpenGL 4.6, so not many GPU's drivers can natively accept and display that... so we need a software conversion.

Reply:
this pretty much needs to happen on any loader that "should" produce a R8_SRGB